src/rhsm/connection.py

```diff
@@ -127,9 +126,124 @@ def __str__(self):
+        restlib_class = restlib_class or Restlib
+        self.host = self.auth_url
+        self.timeout = safe_int(config.get('server', 'server_timeout'))
+        proxy_hostname = None
+        proxy_port = None
+        proxy_user = None
+        proxy_password = None
+        insecure = None
+        self.insecure = insecure
+        if insecure is None:
+            self.insecure = False
+            config_insecure = safe_int(config.get('server', 'insecure'))
+            if config_insecure:
+                self.insecure = True
+
+        self.ca_cert_dir = config.get('rhsm', 'ca_cert_dir')
+        self.ssl_verify_depth = safe_int(config.get('server', 'ssl_verify_depth'))
+
+        # allow specifying no_proxy via api or config
+        no_proxy_override = config.get('server', 'no_proxy')
+        if no_proxy_override:
+            os.environ['no_proxy'] = no_proxy_override
+
+        utils.fix_no_proxy()
+        log.debug('Environment variable NO_PROXY=%s will be used' % no_proxy_override)
+
+        # honor no_proxy environment variable
+        if proxy_bypass(self.host):
+            self.proxy_hostname = None
+            self.proxy_port = None
+            self.proxy_user = None
+            self.proxy_password = None
+        else:
+            info = utils.get_env_proxy_info()
+
+            self.proxy_hostname = proxy_hostname or \
+                config.get('server', 'proxy_hostname') or \
+                info['proxy_hostname']
+            self.proxy_port = proxy_port or \
+                config.get('server', 'proxy_port') or \
+                info['proxy_port']
+            self.proxy_user = proxy_user or \
+                config.get('server', 'proxy_user') or \
+                info['proxy_username']
+            self.proxy_password = proxy_password or \
+                config.get('server', 'proxy_password') or \
+                info['proxy_password']
+
+        url_access_token = self.auth_url + "/realms/" + self.realm + "/protocol/openid-connect/token"
+        self.host = six.moves.urllib.parse.urlparse(url_access_token).hostname or ''
+        url = six.moves.urllib.parse.urlparse(url_access_token).path
+        port = six.moves.urllib.parse.urlparse(url_access_token).port
+        if port:
+            self.ssl_port = port
+        else:
+            self.ssl_port = 443
+        self.handler = ""
+
+        self.conn = restlib_class(self.host, self.ssl_port, self.handler,
+                                  proxy_hostname=self.proxy_hostname, proxy_port=self.proxy_port,
+                                  proxy_user=self.proxy_user, proxy_password=self.proxy_password,
+                                  ca_dir=self.ca_cert_dir, insecure=self.insecure,
+                                  ssl_verify_depth=self.ssl_verify_depth, timeout=self.timeout,
+                                  correlation_id=self.correlation_id, token=self.token)
+
+        #Get access token in exchange for refresh token
```

Comment:
Please add space between # and Get. BTW: do you have PyCharm?

Reply:
Yeah, I use PyCharm